src/EditorFeatures/Core/GoToDefinition/AbstractGoToSymbolService.cs

```diff
@@ -29,12 +35,12 @@ public async Task GetSymbolsAsync(GoToSymbolContext context)
             // We want ctrl-click GTD to be as close to regular GTD as possible.
             // This means we have to query for "third party navigation", from
             // XAML, etc. That call has to be done on the UI thread.
-            var definitions = await Task.Factory.StartNew(() =>
-                GoToDefinitionHelpers.GetDefinitions(symbol, document.Project, thirdPartyNavigationAllowed: true, cancellationToken: cancellationToken)
-                    .WhereAsArray(d => d.CanNavigateTo(document.Project.Solution.Workspace)),
-                        cancellationToken,
-                        TaskCreationOptions.None,
-                        ForegroundTaskScheduler).ConfigureAwait(false);
+            await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(alwaysYield: true, cancellationToken);
```

Comment:
ConfigureAwait?

Reply:
This is a custom struct awaiter; there is no `ConfigureAwait`.